lib/streamlit/elements/lib/subtitle_utils.py

```diff
@@ -0,0 +1,164 @@
+        return False
+
+    # Regular expression to match the SRT timestamp format
+    # It matches the
+    # "hours:minutes:seconds,milliseconds --> hours:minutes:seconds,milliseconds" format
+    timestamp_regex = re.compile(r"\d{2}:\d{2}:\d{2},\d{3} --> \d{2}:\d{2}:\d{2},\d{3}")
+
+    # Split the header into lines and process them
+    lines = header_str.split("\n")
+
+    # Check for the pattern of an SRT file: digit(s), newline, timestamp
+    if len(lines) >= 2 and lines[0].isdigit():
+        match = timestamp_regex.search(lines[1])
+        if match:
+            return True
+
+    return False
+
+
+def srt_to_vtt(srt_data: str | bytes) -> bytes:
+    """
+    Convert subtitles from SubRip (.srt) format to WebVTT (.vtt) format.
+    This function accepts the content of the .srt file either as a string
+    or as a BytesIO stream.
+    Parameters
+    ----------
+    srt_data : str or bytes
+        The content of the .srt file as a string or a bytes stream.
+    Returns
+    -------
+    bytes
+        The content converted into .vtt format.
+    """
+
+    # If the input is a bytes stream, convert it to a string
+    if isinstance(srt_data, bytes):
+        # Decode the bytes to a UTF-8 string
+        try:
+            srt_data = srt_data.decode("utf-8")
+        except UnicodeDecodeError as e:
+            raise ValueError("Could not decode the input stream as UTF-8.") from e
+    if not isinstance(srt_data, str):
+        # If it's not a string by this point, something is wrong.
+        raise TypeError(
+            f"Input must be a string or a bytes stream, not {type(srt_data)}."
+        )
+
+    # Replace SubRip timing with WebVTT timing
+    vtt_data = re.sub(r"(\d{2}:\d{2}:\d{2}),(\d{3})", r"\1.\2", srt_data)
+
+    # Add WebVTT file header
+    vtt_content = "WEBVTT\n\n" + vtt_data
+    # Convert the vtt content to bytes
+    vtt_content = vtt_content.strip().encode("utf-8")
+
+    return vtt_content
+
+
+def process_subtitle_data(
+    coordinates: str,
+    data: str | bytes | io.BytesIO,
+    label: str,
+) -> str:
+    allowed_formats = {".srt", ".vtt"}
+
+    def handle_string_data(data_str: str) -> bytes:
+        """Handles string data, either as a file path or raw content."""
+        if os.path.isfile(data_str):
+            path = Path(data_str)
+            file_extension = path.suffix.lower()
+
+            if file_extension not in allowed_formats:
+                raise ValueError(
+                    f"Incorrect subtitle format {file_extension}. Subtitles must be in "
+                    f"one of the following formats: {', '.join(allowed_formats)}"
+                )
+            with open(data_str, "rb") as file:
+                content = file.read()
+            return srt_to_vtt(content) if file_extension == ".srt" else content
+
+        content_string = data_str.strip()
+
+        if content_string.startswith("WEBVTT"):
+            return content_string.encode("utf-8")
```

Comment:
Per my other [comment](https://github.com/streamlit/streamlit/pull/8057/files#r1489293154), we should support empty strings for both label and URL.

E.g

```py
st.video("example.mp4", subtitles={"":"", "Deutsch": "subtitles-de.vtt", "English": "subtitles-en.srt"})
```

```suggestion
        if content_string.startswith("WEBVTT") or not content_string:
            return content_string.encode("utf-8")
```